docs/developers-guide/substreams-sink-kv.md

```diff
@@ -0,0 +1,231 @@
+
+* The data is written to a key-value store (ex: badger, bigtable, TiKV)
+
+* A Connect-Web interface makes the data available directly from the `substreams-sink-kv` process. Alternatively, you can consume the data directly from your key-value store.
+
+## Requirements
+
+* An existing substreams (including substreams.yaml and rust code) that you want to instrument for `substreams-sink-kv`.
+* (Optional) A key-value store where you want to send your data (a badger local file can be used for development or very-small-scale deployments)
+* Knowledge about substreams development (start [here](https://substreams.streamingfast.io/getting-started/quickstart))
+* Rust installation and compiler
+
+## Installation
+
+* Install [substreams-sink-kv](https://github.com/streamingfast/substreams-sink-kv) latest release
+* Install [substreams](https://github.com/streamingfast/substreams) latest release
+* Install [grpcurl](https://github.com/fullstorydev/grpcurl) to easily read the data back from the KV store
+
+## Quick test of your setup
+
+
+* Before you start instrumenting your substreams for the `kv-sink`, it may be a good idea to make sure that you can run your substreams normally:
+
+```bash
+substreams run -e mainnet.eth.streamingfast.io:443 substreams.yaml some_mapper_module --start-block 1000000 --stop-block +1
+```
+
+> **Note** To connect to a public StreamingFast substreams endpoint, you will need an authentication token, follow this [guide](https://substreams.streamingfast.io/reference-and-specs/authentication) to obtain one.
+
+## Instrumenting your substreams
+
+### Assumptions
+
+The following instructions will assume that you are instrumenting [substreams-eth-block-meta](https://github.com/streamingfast/substreams-eth-block-meta), which contains:
+
+* a store `store_block_meta_end` defined like [this](https://github.com/streamingfast/substreams-eth-block-meta/blob/v0.4.0/substreams.yaml#L29-L34):
+
+```yaml
+# substreams.yaml
+...
+  - name: store_block_meta_end
+    kind: store
+    updatePolicy: set
+    valueType: proto:eth.block_meta.v1.BlockMeta
+    inputs:
+      - source: sf.ethereum.type.v2.Block
+```
+
+* a `eth.block_meta.v1.BlockMeta` protobuf structure like [this](https://github.com/streamingfast/substreams-eth-block-meta/blob/v0.4.0/proto/block_meta.proto#L7-L12):
+```proto
+message BlockMeta {
+  uint64 number = 1;
+  bytes hash = 2;
+  bytes parent_hash = 3;
+  google.protobuf.Timestamp timestamp = 4;
+}
+```
+
+> **Note** The [substreams-eth-block-meta](https://github.com/streamingfast/substreams-eth-block-meta) is already instrumented for sink-kv, the proposed changes here are a simplified version of what has been implemented. Please adjust the proposed code to your own substreams.
+
+### Import the cargo module
+
+1. Add the `substreams-sink-kv` crate to your `Cargo.toml`:
+
+```toml
+# Cargo.toml
+
+[dependencies]
+substreams-sink-kv = "0.1.1"
+# ...
+
+```
+
+2. Add a `kv_out` module  to your `substreams.yaml`:
```

Comment:
All bullet list should be `1.`, Markdown will automatically adjust the number when rendering (revisit all bullet list in the doc)